pkgs/build-support/kernel/make-initrd-ng/src/main.rs

```diff
@@ -48,13 +78,34 @@ impl<T: Clone + Eq + Hash> NonRepeatingQueue<T> {
     }
 }
 
-fn add_dependencies<P: AsRef<Path> + AsRef<OsStr>>(
+fn add_dependencies<P: AsRef<Path> + AsRef<OsStr> + std::fmt::Debug>(
     source: P,
     elf: Elf,
-    queue: &mut NonRepeatingQueue<Box<Path>>,
-) {
+    contents: &[u8],
+    dlopen: &Option<DLOpenConfig>,
+    queue: &mut NonRepeatingQueue<StorePath>,
+) -> eyre::Result<()> {
```

Comment:
The use of mutation and error returns makes me queasy, since this function now can both return an error and mutate its inputs. Since I see that's the pattern with most of the other functions in this file, it's fine, but as a general rule, it's better to do one validation pass (which returns _all_ the errors that could occur, rather than just the first!), then have the rest of the code assert and assume.

Reply:
Such a refactor would be welcome :) I don't think it's strictly necessary for this PR though.